fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+def make_synthetic_dataset(feature_config=None, n_features=20, n_informative=4, random_state=None):
+    """Create a synthetic dataset with a single sensitive feature: 'gender'.
+
+    Parameters
+    ----------
+    feature_config : dict, default=None
+        A dict whose keys are the labels for the gender feature, and whose set
+        the :code:`n_samples` and :code:`class_sep` parameters for :code:`make_classification`.
+        If `None`, the following dictionary is used by default::
+
+            {
+                'Man': {'n_samples': 500, 'class_sep': 1},
+                'Other': {'n_samples': 500, 'class_sep': 0.5},
+                'Unspecified': {'n_samples': 500, 'class_sep': 0.5},
+                'Woman': {'n_samples': 500, 'class_sep': 2},
+            }
+
+    n_features : int, default=20
+        The total number of features.
+
+    n_informative : int, default=4
+        The number of informative features.
```

Comment:
A little more explanation might be nice here, even if it's mainly just passing through to SciKit. I'm guessing this means that four of the features will have some effect on the prediction, but the other 16 will be completely random?

Reply:
I was thinking about how to explain in this in light of the bias that we add by flipping classes, but the problem is that degrades these informative features. Rather, we should leave the labels alone, and rearrange the sensitive feature assignments.

This is actually much easier, because `make_classification` accepts a `weights` kwarg that lets you set the proportion of positive labels. Update incoming...